### PR TITLE
Add scroll button and auto scroll features to Column

### PR DESCRIPTION
**Pull request by ahuang11**
Adds `scroll_arrow_threshold` and `auto_scroll` to the newly subclassed Column

Improves `scroll=True` so it only shows scroll in the direction the widget is going (Column -> Vertical / Y, Row -> Horizontal / X).

- [x] Test
- [x] Docs
- [ ] Ascending -- no longer planning to do this in this PR

### CODE REVIEW

**Comment by ahuang11** [comment]
Encountering problem with it scrolling to second latest element rather than latest.
image

Also, sticky scroll for `column-reverse` is weirdly located, and changing `top` doesn't move it.
image

**Comment by ahuang11** [comment]
Scrolling too early fixed by using requestAnimationFrame

**Comment by ahuang11** [comment]
I found a solution to `ascending` and sticky arrow like [this](https://discourse.holoviz.org/t/why-doesnt-sticky-work-with-column-reverse/5689/2), but I don't think it's a good idea to nest the child objects in yet another div container, especially for a fundamental layout object like Column.

So maybe in another PR for another feature or someone else can find a more elegant solution. For now, I think this is good enough.

**Comment by ahuang11** [comment]
Wondering why all the other layouts are being affected by `auto_scroll`

```

___________________________ test_card_get_root_title ___________________________
[gw0] linux -- Python 3.8.17 /usr/share/miniconda3/envs/test-environment/bin/python3.8

document = <bokeh.document.document.Document object at 0x7fb044a47040>
comm = Comm(id='9e000616e3d247f9bbc65b8b8afcff42', name='Comm13449')

    def test_card_get_root_title(document, comm):
        div1 = Div()
        div2 = Div()
        layout = Card(div1, div2, title='Test')
    
>       model = layout.get_root(document, comm=comm)

panel/tests/layout/test_card.py:43: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
panel/layout/base.py:306: in get_root
    root = super().get_root(doc, comm, preprocess)
panel/viewable.py:658: in get_root
    root = self._get_model(doc, comm=comm)
panel/layout/base.py:178: in _get_model
    model.update(**props)
/usr/share/miniconda3/envs/test-environment/lib/python3.8/site-packages/bokeh/core/has_props.py:474: in update
    setattr(self, k, v)
/usr/share/miniconda3/envs/test-environment/lib/python3.8/site-packages/bokeh/core/has_props.py:333: in __setattr__
    self._raise_attribute_error_with_matches(name, properties)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Card(id='p3248', ...), name = 'auto_scroll'
properties = {'active_header_background': Nullable(String), 'align': Either(Auto, Enum('start', 'center', 'end'), Tuple(Enum('start..., Enum('start', 'center', 'end'))), 'aspect_ratio': Either(Null, Auto, Float), 'button_css_classes': List(String), ...}

    def _raise_attribute_error_with_matches(self, name: str, properties: Iterable[str]) -> NoReturn:
        matches, text = difflib.get_close_matches(name.lower(), properties), "similar"
    
        if not matches:
            matches, text = sorted(properties), "possible"
    
>       raise AttributeError(f"unexpected attribute {name!r} to {self.__class__.__name__}, {text} attributes are {nice_join(matches)}")
E       AttributeError: unexpected attribute 'auto_scroll' to Card, possible attributes are active_header_background, align, aspect_ratio, button_css_classes, children, collapsed, collapsible, context_menu, css_classes, disabled, flow_mode, header_background, header_color, header_css_classes, header_tag, height, height_policy, hide_header, js_event_callbacks, js_property_callbacks, margin, max_height, max_width, min_height, min_width, name, resizable, sizing_mode, spacing, styles, stylesheets, subscribed_events, syncable, tag, tags, visible, width or width_policy

```

File: panel/layout/base.py [hunk 1]
```diff
@@ -879,12 +886,29 @@ class Column(ListPanel):
     >>> pn.Column(some_widget, some_pane, some_python_object)
     """
 
-    _bokeh_model: ClassVar[Type[Model]] = BkColumn
+    auto_scroll = param.Boolean(
```
Comment (philippjfr) [inline panel/layout/base.py:889]: Please change to match the existing code formatting style.

File: panel/layout/base.py [hunk 2]
```diff
@@ -879,12 +886,29 @@ class Column(ListPanel):
...
+    _bokeh_model: ClassVar[Type[Model]] = PnColumn
 
     _direction = 'vertical'
 
     _stylesheets: ClassVar[list[str]] = [f'{CDN_DIST}css/listpanel.css']
 
+    def __init__(self, *objects: Any, **params: Any):
+        if "auto_scroll" in params or "scroll_arrow_threshold" in params:
+            params["scroll"] = True
+        super().__init__(*objects, **params)
```
Comment (philippjfr) [inline panel/layout/base.py:910]:
```suggestion
    @param.depends("auto_scroll", "scroll_arrow_threshold", watch=True, on_init=True)
    def _set_scrollable(self):
        self.scroll = self.scroll or self.auto_scroll or self.scroll_arrow_threshold
        
```

File: panel/layout/base.py [hunk 3]
```diff
@@ -879,12 +886,29 @@ class Column(ListPanel):
     >>> pn.Column(some_widget, some_pane, some_python_object)
     """
 
-    _bokeh_model: ClassVar[Type[Model]] = BkColumn
+    auto_scroll = param.Boolean(
+        default=False,
+        doc="Whether to scroll to the latest object on update."
+    )
+
+    scroll_arrow_threshold = param.Number(
```
Comment (philippjfr) [inline panel/layout/base.py:894]: I'd call this `scroll_button_threshold` and also please set `bounds=(0, None)`.

**Comment by philippjfr** [comment]
Pushed a number of fixes here:

1. Conditionally adding callbacks in `connect_signals` is never a good idea. Generally simply attach the callback and skip it if needed, this means that if the threshold is changed later the callback is attached correctly.
2. The CSS class on the button was `scroll-button` but the CSS still called it `scroll-down-arrow`.
3. You had a `this.child_views.slice(1)` in there which meant it would skip rendering the first child entirely
4. In general you should not have to access the value via the property, instead you just access it directly on the model. I am now getting some compiler warnings that those properties aren't defined which indicate something is wrong. I haven't found what that is yet.

The arrow is currently causing the children to be shifted down slightly. Haven't found a fix for that yet.

**Comment by ahuang11** [comment]
I fixed the compiling issue by reverting to using `get_value()` based on this traceback:
```

panel/models/column.ts:31:44 - error TS2367: This comparison appears to be unintentional because the types 'Property<number>' and 'number' have no overlap.

31     console.log(scrollDistanceFromBottom, (threshold !== 0) && (scrollDistanceFromBottom >= threshold))
                                              ~~~~~~~~~~~~~~~
panel/models/column.ts:31:65 - error TS2365: Operator '>=' cannot be applied to types 'number' and 'Property<number>'.

31     console.log(scrollDistanceFromBottom, (threshold !== 0) && (scrollDistanceFromBottom >= threshold))
                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
panel/models/column.ts:33:19 - error TS2367: This comparison appears to be unintentional because the types 'Property<number>' and 'number' have no overlap.

33       "visible", (threshold !== 0) && (scrollDistanceFromBottom >= threshold)
                     ~~~~~~~~~~~~~~~
panel/models/column.ts:33:40 - error TS2365: Operator '>=' cannot be applied to types 'number' and 'Property<number>'.

33       "visible", (threshold !== 0) && (scrollDistanceFromBottom >= threshold)
```

However, I can't get the scrollbar to show up now (maybe cache issue).

File: panel/models/column.ts [hunk 4]
```diff
@@ -26,9 +26,8 @@ export class ColumnView extends BkColumnView {
   }
 
   toggle_scroll_arrow(): void {
-    const threshold = this.model.scroll_button_threshold
+    const threshold = this.model.properties.scroll_button_threshold.get_value()
```
Comment (philippjfr) [inline panel/models/column.ts:29]: Please revert.

File: panel/models/column.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ export class ColumnView extends BkColumnView {
   }
 
   scroll_to_latest(): void {
-    if (!this.model.auto_scroll)
+    if (!this.model.properties.auto_scroll)
```
Comment (philippjfr) [inline panel/models/column.ts:19]: Please revert.

**Comment by philippjfr** [comment]
Let's figure out why it was giving compiler errors rather than introducing workarounds.

**Comment by ahuang11** [comment]
Oh I didn't consider this as a workaround, but I suppose I'm unfamiliar with Typescript. I'll study card.ts to see how this is done.

**Comment by philippjfr** [comment]
Fixed the compilation errors and reverted your changes.

**Comment by philippjfr** [comment]
pre-commit.ci autofix

**Comment by ahuang11** [comment]
Oops, I just saw your change too.

**Comment by ahuang11** [comment]
pre-commit.ci autofix

**Comment by philippjfr** [comment]
The circle does not look square though, not sure why but I was only able to fix that by setting a `min-height`

**Comment by ahuang11** [comment]
Okay I fixed the CSS so it doesn't touch the bottom of the screen and used min height
image

**Comment by philippjfr** [comment]
UI tests still failing and the scroll button doesn't seem to appear on initial render.

**Comment by ahuang11** [comment]
Okay, I believe the widgetbox is related, but unsure if the jupyterlite is flaky / related.

```
FAILED panel/tests/ui/layout/test_widgetbox.py::test_widgetbox_vertical_scroll - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
=========================== logs ===========================
waiting for locator(".bk-Column")
============================================================
FAILED panel/tests/ui/io/test_jupyterlite.py::test_jupyterlite_execution - playwright._impl._api_types.TimeoutError: Timeout 120000ms exceeded.
=========================== logs ===========================
waiting for locator(".noUi-handle")
```

File: panel/tests/ui/layout/test_column.py [hunk 6]
```diff
@@ -21,7 +21,186 @@ def test_column_scroll(page, port):
...
+    assert bbox['height'] == 420
+
+    assert 'scrollable-vertical' in column.get_attribute('class')
+
+    # assert scroll button is invisible on render
+    scroll_arrow = page.locator(".scroll-button")
+    assert scroll_arrow.get_attribute('class') == 'scroll-button'
+    assert not scroll_arrow.is_visible()
+
+    # assert scroll button is visible beyond threshold
```
Comment (ahuang11) [inline panel/tests/ui/layout/test_column.py:175]:
```suggestion
    # assert scroll button is still invisible beyond threshold
```

**Comment by ahuang11** [comment]
I think this is ready; can't tell if the error is related to this PR.
```
=========================== short test summary info ============================
FAILED panel/tests/ui/io/test_jupyterlite.py::test_jupyterlite_execution - playwright._impl._api_types.TimeoutError: Timeout 120000ms exceeded.
=========================== logs ===========================
waiting for locator(".noUi-handle")
============================================================
===== 1 failed, 363 passed, 33 xfailed, 311 warnings in 837.07s (0:13:57) ======
TaskFailed - taskid:test_ui
Command failed: 'pytest "panel" "--cov=./panel" "--cov-report=xml" "--cov-config=.uicoveragerc" "--ui" "--jupyter" "--browser" "chromium" "-n" "logical" "--dist" "loadgroup"' returned 1
```

**Comment by ahuang11** [comment]
Ooh I think https://github.com/holoviz/panel/pull/5312 might fix the UI issue.

**Comment by ahuang11** [comment]
Okay maybe not
```

=========================== short test summary info ============================
FAILED panel/tests/ui/io/test_jupyterlite.py::test_jupyterlite_execution - playwright._impl._api_types.TimeoutError: Timeout 120000ms exceeded.
=========================== logs ===========================
waiting for locator(".noUi-handle")
============================================================

```

**Review by philippjfr — APPROVED** [review]
Thanks, looks good!

**Pull request merged**